File: init_settings.py
#!/usr/bin/env python3
"""
Инициализация настроек бота в Supabase
"""

import os
from supabase import create_client
from dotenv import load_dotenv

# Загрузка переменных окружения
load_dotenv()

SUPABASE_URL = os.getenv('SUPABASE_URL')
SUPABASE_KEY = os.getenv('SUPABASE_KEY')

def init_settings():
    """Инициализация настроек бота"""
    print("🔧 Инициализация настроек бота...")
    
    if not SUPABASE_URL or not SUPABASE_KEY:
        print("❌ Отсутствуют переменные окружения SUPABASE_URL или SUPABASE_KEY")
        return False
    
    try:
        client = create_client(SUPABASE_URL, SUPABASE_KEY)
        
        # Проверяем существование настроек
        result = client.table('settings').select('*').eq('id', 1).execute()
        
        if result.data:
            print("✅ Настройки уже существуют")
            print(f"📞 Телефон: {result.data[0].get('sbp_phone', 'не настроен')}")
            print(f"🏦 Банк: {result.data[0].get('sbp_bank', 'не настроен')}")
            return True
        
        # Создаем настройки по умолчанию
        settings_data = {
            'id': 1,
            'sbp_phone': '79123456789',
            'sbp_bank': 'Сбер',
            'sbp_recipient': 'Lavka26',
            'promo_price': 199.00
        }
        
        result = client.table('settings').insert(settings_data).execute()
        
        if result.data:
            print("✅ Настройки успешно созданы:")
            print(f"📞 Телефон: {settings_data['sbp_phone']}")
            print(f"🏦 Банк: {settings_data['sbp_bank']}")
            print(f"👤 Получатель: {settings_data['sbp_recipient']}")
            print(f"💰 Цена рекламы: {settings_data['promo_price']} ₽")
            return True
        else:
            print("❌ Ошибка создания настроек")
            return False
            
    except Exception as e:
        print(f"❌ Ошибка при инициализации настроек: {e}")
        return False

if __name__ == "__main__":
    success = init_settings()
    if success:
        print("\n🎉 Настройки бота успешно инициализированы!")
    else:
        print("\n💥 Ошибка инициализации настроек")

File: init_settings.js
const { createClient } = require('@supabase/supabase-js');

// Конфигурация
const SUPABASE_URL = process.env.SUPABASE_URL || 'https://your-project.supabase.co';
const SUPABASE_KEY = process.env.SUPABASE_KEY || 'your-anon-key';

const supabase = createClient(SUPABASE_URL, SUPABASE_KEY);

async function initSettings() {
    try {
        console.log('🔧 Инициализация таблицы settings...');
        
        // Пробуем создать таблицу через SQL
        const { error: createError } = await supabase.rpc('exec_sql', {
            sql: `
                CREATE TABLE IF NOT EXISTS settings (
                    id SERIAL PRIMARY KEY,
                    sbp_phone VARCHAR(20),
                    sbp_bank VARCHAR(255),
                    created_at TIMESTAMP WITH TIME ZONE DEFAULT NOW(),
                    updated_at TIMESTAMP WITH TIME ZONE DEFAULT NOW()
                );
                
                INSERT INTO settings (id, sbp_phone, sbp_bank) 
                SELECT 1, NULL, NULL 
                WHERE NOT EXISTS (SELECT 1 FROM settings WHERE id = 1);
            `
        });
        
        if (createError) {
            console.log('❌ Ошибка создания таблицы через RPC:', createError);
            
            // Пробуем прямой подход - просто вставляем запись
            console.log('🔄 Пробуем прямой подход...');
            const { data, error: insertError } = await supabase
                .from('settings')
                .upsert({ 
                    id: 1, 
                    sbp_phone: null, 
                    sbp_bank: null 
                }, {
                    onConflict: 'id'
                });
                
            if (insertError) {
                console.log('❌ Ошибка прямого подхода:', insertError);
                
                // Последний вариант - создаем через raw SQL
                console.log('🔄 Пробуем через raw SQL...');
                const { error: rawError } = await supabase
                    .from('settings')
                    .select('*')
                    .limit(1);
                    
                if (rawError && rawError.code === 'PGRST116') {
                    console.log('❌ Таблица settings не существует. Нужно создать вручную в Supabase Dashboard');
                    console.log('📝 SQL для создания:');
                    console.log(`
CREATE TABLE settings (
    id SERIAL PRIMARY KEY,
    sbp_phone VARCHAR(20),
    sbp_bank VARCHAR(255),
    created_at TIMESTAMP WITH TIME ZONE DEFAULT NOW(),
    updated_at TIMESTAMP WITH TIME ZONE DEFAULT NOW()
);

INSERT INTO settings (id, sbp_phone, sbp_bank) 
VALUES (1, NULL, NULL);

ALTER TABLE settings DISABLE ROW LEVEL SECURITY;
                    `);
                } else {
                    console.log('✅ Таблица settings существует!');
                }
            } else {
                console.log('✅ Таблица settings инициализирована!');
            }
        } else {
            console.log('✅ Таблица settings создана через RPC!');
        }
        
        // Проверяем, что таблица работает
        const { data: testData, error: testError } = await supabase
            .from('settings')
            .select('*')
            .eq('id', 1)
            .single();
            
        if (testError) {
            console.log('❌ Ошибка проверки таблицы:', testError);
        } else {
            console.log('✅ Таблица settings работает корректно:', testData);
        }
        
    } catch (error) {
        console.error('❌ Общая ошибка:', error);
    }
}

initSettings();
